hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/CacheConfig.java

```diff
@@ -398,6 +420,21 @@ public ByteBuffAllocator getByteBuffAllocator() {
     return this.byteBuffAllocator;
   }
 
+  private long getCacheCompactedBlocksOnWriteThreshold(Configuration conf) {
+    long cacheCompactedBlocksOnWriteThreshold = conf.getLong(CACHE_COMPACTED_BLOCKS_ON_WRITE_THRESHOLD_KEY,
+      DEFAULT_CACHE_COMPACTED_BLOCKS_ON_WRITE_THRESHOLD);
+
+    if (cacheCompactedBlocksOnWriteThreshold < 0) {
+      String warnMessage =
+        "cacheCompactedBlocksOnWriteThreshold value :" + cacheCompactedBlocksOnWriteThreshold
+          + "is less than 0, resetting it to: " + DEFAULT_CACHE_COMPACTED_BLOCKS_ON_WRITE_THRESHOLD;
+      LOG.warn(warnMessage);
```

Comment:
nit : May be just add the string directly in Log rather than in msg? Generally they don't do this way in hbase code unless the msg is very big. Minor only.